src/engine/Filter.cpp

```diff
@@ -45,24 +44,59 @@ string Filter::getDescriptor() const {
 // _____________________________________________________________________________
 ProtoResult Filter::computeResult(bool requestLaziness) {
   LOG(DEBUG) << "Getting sub-result for Filter result computation..." << endl;
-  std::shared_ptr<const Result> subRes = _subtree->getResult(requestLaziness);
+  std::shared_ptr<const Result> subRes = _subtree->getResult(true);
   LOG(DEBUG) << "Filter result computation..." << endl;
   checkCancellation();
 
+  auto localVocab = subRes->getSharedLocalVocab();
   if (subRes->isFullyMaterialized()) {
     IdTable result = filterIdTable(subRes, subRes->idTable());
     LOG(DEBUG) << "Filter result computation done." << endl;
 
-    return {std::move(result), resultSortedOn(), subRes->getSharedLocalVocab()};
+    return {std::move(result), resultSortedOn(), std::move(localVocab)};
   }
-  auto localVocab = subRes->getSharedLocalVocab();
-  return {[](auto subRes, auto* self) -> cppcoro::generator<IdTable> {
-            for (IdTable& idTable : subRes->idTables()) {
-              IdTable result = self->filterIdTable(subRes, idTable);
-              co_yield result;
-            }
-          }(std::move(subRes), this),
-          resultSortedOn(), std::move(localVocab)};
+  if (requestLaziness) {
+    return {[](auto subRes, auto* self) -> cppcoro::generator<IdTable> {
+              for (IdTable& idTable : subRes->idTables()) {
+                IdTable result = self->filterIdTable(subRes, idTable);
+                co_yield result;
+              }
+            }(std::move(subRes), this),
+            resultSortedOn(), std::move(localVocab)};
+  }
+
+  IdTable result = ad_utility::callFixedSize(
+      getSubtree().get()->getResultWidth(),
+      [this, subRes = std::move(subRes)]<int WIDTH>() {
+        IdTable result{WIDTH, getExecutionContext()->getAllocator()};
+        IdTableStatic<WIDTH> output =
+            std::move(result).toStatic<static_cast<size_t>(WIDTH)>();
+
+        std::vector<ColumnIndex> sortedOn = subRes->sortedBy();
+
+        for (IdTable& idTable : subRes->idTables()) {
+          sparqlExpression::EvaluationContext evaluationContext(
+              *getExecutionContext(), _subtree->getVariableColumns(), idTable,
+              getExecutionContext()->getAllocator(), subRes->localVocab(),
+              cancellationHandle_, deadline_);
+
+          // TODO<joka921> This should be a mandatory argument to the
+          // EvaluationContext constructor.
+          evaluationContext._columnsByWhichResultIsSorted = std::move(sortedOn);
+
+          computeFilterImpl(output, evaluationContext);
+
+          // Reuse vector
+          sortedOn = std::move(evaluationContext._columnsByWhichResultIsSorted);
+          checkCancellation();
+        }
+
+        return std::move(output).toDynamic();
+      });
+
```

Comment:
Isn't a much simpler implementation to do the following:

Implement `filterIdTableAndAppend(SameArgs|AsfilterIdTable, outputTable)`
(simple, almost the same as `filterIdTable`,
Then implement `filterIdTable` in terms of this helper, and then the newly added case here becomes a simple
```
IdTable result = ...
for (auto table : lazySubresult) {
  filterAndAppend(table, result);
}
return tableAndWhatNotElse

```

Reply:
1. This fucntion stays a rather simple switch
2. The complicated setting up of the evaluationClontext etc. stays in one place (better for archtiecture/fixing bugs etc.

Reply:
(The rest of the changes I will look at once this is done, because they probably will be affected.